backend_skeleton/sensor_methods.py

```diff
@@ -0,0 +1,257 @@
+    # but usually can be ignored. If it happens more than a few times over
+    # the span of a few seconds, the sensor needs to be reset.
+    error_count = 0
+    start_time = time.time()
+    sensor_data = []
+    interpolated_data = []
+    while True:
+        current_time = time.time()
+        time_elapsed = current_time - start_time
+        # Check for new data, available every 2 seconds
+        try:
+            if scd.data_available:  # If fresh data is available, get it
+                error_count = 0
+                interval_data = get_interval_data(scd, time_elapsed)
+                sensor_data.append(interval_data)
+                if debug_live_data:  # print raw data to screen to debug sensor
+                    output_to_screen(interval_data)
+        except RuntimeError as e:
+            # Occasionally sensor does not want to respond
+            # Ordinarily this should be I2C read error: max entries reached
+            print(e)  # Print the error
+            error_count += 1
+            if error_count > 10:
+                print("SENSOR FAILURE... killing program")
+                sys.exit() # Kill the program because something bad is happening
+
+        # Check if there is data to interpolate
+        while (len(sensor_data) > 1 and
+           sensor_data[-1]['seconds'] > interpolated_time + desired_time_step):
+            # Get the smallest pair without going under
+            for index, data_set in enumerate(reversed(sensor_data)):
+                if data_set['seconds'] > interpolated_time + desired_time_step:
+                    smallest_set_without_going_under = data_set
+                if data_set['seconds'] < interpolated_time + desired_time_step:
+                    break
+
+            #Get the largest pair without going over
+            largest_set_without_going_over = sensor_data[-(index + 1)]
+            # Interpolate to find the new value
+            new_time = interpolated_time + desired_time_step
+            interpolated_set = interpolate_linear(
+                new_time,
+                # Time and values of the largest pair
+                largest_set_without_going_over,
+                # Time and values of the smallest pair
+                smallest_set_without_going_under
+            )
+            interpolated_data.append(interpolated_set)
+            if debug_interpolated_data:
+                 # print new data to screen to debug sensor
+                output_to_screen(interpolated_set)
+            interpolated_time += desired_time_step
+        
+        # If more than WRITE_QTY data points have been gathered, write to file
+        if len(interpolated_data) > WRITE_QTY:
+
+            column_names = ['time', 'co2','temp','humidity']
+            file_name = f"scd_data/raw/raw_data{interpolated_time}.csv"
+            #Write Raw Data
+            with open(file_name, 'w', newline='') as csvfile:
```

Comment:
In theory here you want to do something like:
```suggestion
            column_names = ['time', 'co2', 'temp', 'humidity']
            file = Path('scd_data', 'raw', f'raw_data{interpolated_time}.csv')
            # write Raw Data
            with file.open('w', newline='') as csvfile:
```
To avoid hardcoding the linux path separator.

In addition, if  you set the base dir as e.g. `base_dir = Path('scd_data')`, then you can just do:

```py
file = base_dir / 'raw' / f'raw_data{interpolated_time}.csv'
```